clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -1082,6 +1098,14 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
 
     if (isa<SYCLIntelFPGANofusionAttr>(A))
       setSYCLNofusionEnable();
+
+    if (const auto *IntelFpgaPipeline =
+            dyn_cast<SYCLIntelFpgaPipelineAttr>(A)) {
+      setSYCLIntelFPGAPipelineEnable();
+      setSYCLIntelFPGANPipelines(IntelFpgaPipeline->getValue()
+                                     ->getIntegerConstantExpr(Ctx)
+                                     ->getSExtValue());
```

Comment:
Isn't the value already known to be a constant expression by this point, so we can cast to `ConstantExpr` and get the result directly?